### DISCUSSION TITLE
Debounce webhook execution with flows?

### DISCUSSION THREAD

**Discussion by m1212e** [directus/directus#16006] [Archive]
Is there a way to debounce webhooks with flows?

For example if I edit a page, save it and then edit another, I don't want a webhook to be sent twice but x amount of seconds after the second edit happened. Is this possible?

**Comment by rijkvanzanten** [comment]
There's currently no built-in way to do that. That being said, I'd love to hear some more thoughts on how you'd think it should ideally work. Would you imagine that it "combines" all the edits and then sends one webhook with all the payloads, or would it skip over the first couple edits completely and then only send the last webhook payload?

**Reply by m1212e** [reply]
Since I only use it to start a build process I don't really care about the body because I don't use it anyway.
Thinking about your question, it would probably make most sense to discard all but the last one. At least that's my understanding of a debouncer.

**Reply by rijkvanzanten** [reply]
Makes sense for that use case! Just thinking a step ahead here for the cases where you _do_ care about the payload, in which case you can't just ignore the previous runs, but instead have to concatenate them (somehow) so you can still properly respond to the actual payloads you're reacting to 🤔

**Reply by m1212e** [reply]
In that case a concatenation would make sense.

**Reply by connorwinston** [reply]
This would actually be really useful, I switched our deploys from web hooks to manual flow triggers because of the many deploys it was triggering. Would be nice to set a number of seconds or minutes that it "collects" the changes after a change, then collect the changes, and then post them all in one request. If there was only one change after the amount of time it could send an array with 1 item.

**Reply by rijkvanzanten** [reply]
Yeah! Similarly, I'd like to think through the ability for request operations to automatically retry with exponential backoff on failures

**Reply by connorwinston** [reply]
That would be useful as well and would make integrating Directus with other services more robust!

**Reply by benhaynes** [reply]
Cloud extension operation, please ☺️

**Comment by northern-mindset** [comment]
> I only use it to start a build process

Regarding this particular use case, namely debouncing the website build, this can be addressed on the build pipeline side.

See [similar question on SO](https://stackoverflow.com/a/76571677/3082178), I'm posting my answer here as well:

For example, [Workflow Run Debounce action](https://github.com/marketplace/actions/workflow-run-debounce) for GitHub.

Another approach is to have a middleware like https://hookbox.freighter.studio/ to debounce the webhook requests in between the Headless CMS and the Build Pipeline. Probably there are open-source and self-hosted alternatives.
